library/Fission/User/Creator.hs

```diff
@@ -1,7 +1,159 @@
 module Fission.User.Creator
   ( module Fission.User.Creator.Class
   , module Fission.User.Creator.Error
+  , createDB
+  , createWithPasswordDB
+  , createWithHerokuDB
   ) where
 
-import Fission.User.Creator.Class
-import Fission.User.Creator.Error
+import           Fission.User.Creator.Class
+import           Fission.User.Creator.Error
+
+import           Data.UUID (UUID)
+import           Database.Esqueleto hiding ((<&>))
+import           Servant
+
+import           Fission.Prelude
+import           Fission.Error as Error
+import           Fission.Models
+
+import           Fission.Key as Key
+
+import qualified Fission.Platform.Heroku.Region.Types  as Heroku
+import qualified Fission.Platform.Heroku.AddOn.Creator as Heroku.AddOn
+
+import qualified Fission.User.Creator.Error as User
+import           Fission.User.Password      as Password
+import           Fission.User.Types
+import qualified Fission.User.Validation    as User
+
+import           Network.IPFS.Bytes.Types
+
+import qualified Fission.App.Content as App.Content
+
+
+createDB ::
+     MonadIO m
+  => Username
+  -> Key.Public
+  -> Email
+  -> UTCTime
+  -> Transaction m (Either Errors UserId)
+createDB username pk email now =
+  User
+    { userPublicKey     = Just pk
+    , userUsername      = username
+    , userEmail         = Just email
+    , userRole          = Regular
+    , userActive        = True
+    , userHerokuAddOnId = Nothing
+    , userSecretDigest  = Nothing
+    , userDataRoot      = App.Content.empty
+    , userDataRootSize  = Bytes 0
+    , userInsertedAt    = now
+    , userModifiedAt    = now
+    }
+    |> User.check
+    |> \case
+      Left err ->
+        return $ Error.openLeft err
+
+      Right user ->
+        insertUnique user >>= \case
+          Just userId -> return $ Right userId
+          Nothing -> determineConflict username (Just pk)
+
+
+createWithPasswordDB ::
+     MonadIO m
+  => Username
+  -> Password
+  -> Email
+  -> UTCTime
+  -> Transaction m (Either Errors UserId)
+createWithPasswordDB username password email now =
+  Password.hashPassword password >>= \case
+    Left err ->
+      return $ Error.openLeft err
+
+    Right secretDigest ->
+      User
+        { userPublicKey     = Nothing
+        , userUsername      = username
+        , userEmail         = Just email
+        , userRole          = Regular
+        , userActive        = True
+        , userHerokuAddOnId = Nothing
+        , userSecretDigest  = Just secretDigest
+        , userDataRoot      = App.Content.empty
+        , userDataRootSize  = Bytes 0
+        , userInsertedAt    = now
+        , userModifiedAt    = now
+        }
+        |> insertUnique
+        |> bind \case
+          Just userId -> return $ Right userId
+          Nothing -> determineConflict username Nothing
+
+
+createWithHerokuDB ::
```

Comment:
How likely are we to use this funtion in multiple places? Why not just stick it right in `Fission.Types`?

Reply:
If it’s “that file is getting long” — this is another case of “when we break up the subprojects, we can use less abstraction, and thus less in that file”

Reply:
yeah it was "this file is getting long" & I liked maintaining that separation of "database" & "everything else"

I do use `createDB` in `server/Main` to ensure the default user is created since we're not in `Fission` yet